src/newcharacter.cpp

```diff
@@ -394,15 +395,21 @@ void avatar::randomize( const bool random_scenario, points_left &points, bool pl
 
 void avatar::add_profession_items()
 {
+    // Our profession should not be a hobby
+    if( prof->is_hobby() ) {
+        return;
+    }
+
     std::list<item> prof_items = prof->items( male, get_mutations() );
 
     for( item &it : prof_items ) {
         if( it.has_flag( STATIC( flag_id( "WET" ) ) ) ) {
             it.active = true;
             it.item_counter = 450; // Give it some time to dry off
         }
+
         // TODO: debugmsg if food that isn't a seed is inedible
-        if( it.has_flag( json_flag_no_auto_equip ) ) {
+        else if( it.has_flag( json_flag_no_auto_equip ) ) {
```

Comment:
Why adding the else here? Items with the WET flag still needs to be handled according to the no-auto-equip/auto-wield flags.

Reply:
oh whoops, that was left over from when I tried to integrate hobbies into add_profession_items(). I ended up doing add_hobby_items() instead, and left the else in. Good catch!